dashboard/modules/job/job_manager.py

```diff
@@ -234,8 +234,8 @@ def _exec_entrypoint(self, logs_path: str) -> subprocess.Popen:
                 stderr=subprocess.STDOUT,
                 # Ray intentionally blocks SIGINT in all processes, so if the user wants
                 # to stop job through SIGINT, we need to unblock it in the child process
-                preexec_fn=lambda: signal.pthread_sigmask(
-                    signal.SIG_UNBLOCK, {signal.SIGINT}
+                preexec_fn=(
+                    lambda: signal.pthread_sigmask(signal.SIG_UNBLOCK, {signal.SIGINT})
                 )
```

Comment:
Could you explain why adding parentheses fixes this?

Reply:
The reason has to do with what gets assigned to the `preexec_fn` function. Without the parentheses you will always assign the lambda and the if-statement gets executed as part of the lambda call. With parenthesis you will only assign the lambda when you're on a non-Windows system and on Windows systems you assign None.
On Unix systems this is not an issue due to it allowing a `preexec_fn` method and as such it will not crash before even attempting to execute the lamba.

The following short script shows the error/fix.
```
import subprocess, sys

print("Before 1")
subprocess.Popen(
    "cmd.exe /c echo Hello World",
    preexec_fn=(lambda: print('Hello World')) if sys.platform != "win32"  else None
)
print("Before 2")
subprocess.Popen(
    "cmd.exe /c echo Hello World",
    preexec_fn=lambda: print('Hello World') if sys.platform != "win32"  else None
)
```

Output (using a modified subprocess.py file to include the additional prints):
```
Before 1
>> "Value assigned to 'preexec':  None
Hello World
Before 2
>> "Value assigned to 'preexec':  <function <lambda> at 0x0000027EAE8E3D90>
Traceback (most recent call last):
  File "C:\temp.py", line 10, in <module>
    subprocess.Popen(
  File "<python path>\lib\subprocess.py", line 792, in __init__
    raise ValueError("preexec_fn is not supported on Windows "
ValueError: preexec_fn is not supported on Windows platforms: <function <lambda> at 0x0000027EAE8E3D90>
```

Reply:
Oh oof I totally missed the if-statement. Thanks for the detailed explanation and fix!

Reply:
Yeah, it was an interesting debug experience for us, we overlooked it for a long time ourselves 😬